### ISSUE TITLE
Module not found inside getNetworkServiceNames

### ISSUE THREAD

**Issue by saphitv** [oracle/node-oracledb#1706]
## Versions
oracledb: 6.7.0
node: 21.4.0
platform: win32
arch: x64

## Description
I'm encountering issues launching my Next.js (15.0.3) project. It crashes upon startup.
I tried to change the require from an absolute path to `./thin/sqlnet/paramParser.js` and it works.
I am using oracledb in thin mode.
I was wondering if i was doing something wrong or what the solution might be. 

If the information is not enough, I remain at your disposal.

### Stacktrace
```
 ⨯ ./node_modules/oracledb/lib/oracledb.js:668:47
Module not found: Can't resolve 'thin/sqlnet/paramParser.js'
  666 | //-----------------------------------------------------------------------------
  667 | async function getNetworkServiceNames(configDir) {
> 668 |   const { NLParamParser, tnsnamesFilePath } = require('thin/sqlnet/paramParser.js');
      |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  669 |   const nlParamParser =  new NLParamParser;
  670 |   const filePath = tnsnamesFilePath(configDir);
  671 |   const aliasht = await nlParamParser.initializeNlpa(filePath);
```

**Comment by sharadraju** [comment]
Thanks for reporting this.
I believe your fix is one of the right solutions.

Is it happening only on startup of your Next.js project?

**Comment by sharadraju** [comment]
Ok, I am able to replicate this issue internally.
@saphitv 
We will add the fix that you had mentioned in the next patch release:
```
diff --git a/lib/oracledb.js b/lib/oracledb.js
index ea3b41528..b71690b82 100644
--- a/lib/oracledb.js
+++ b/lib/oracledb.js
@@ -665,7 +665,7 @@ async function createPool(options) {
 // If a tnsnames.ora file does not exist, then an exception is raised
 //-----------------------------------------------------------------------------
 async function getNetworkServiceNames(configDir) {
-  const { NLParamParser, tnsnamesFilePath } = require('thin/sqlnet/paramParser.js');
+  const { NLParamParser, tnsnamesFilePath } = require('./thin/sqlnet/paramParser.js');
   const nlParamParser =  new NLParamParser;
   const filePath = tnsnamesFilePath(configDir);
   const aliasht = await nlParamParser.initializeNlpa(filePath);
```

As a temporary workaround, you can add also add the following change in your next.config.(m)js file:

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack: config => {
    /**
     * These packages need to be added as external, else Oracle DB will try to load them due to a
     * Webpack bug.
     *
     * See these two issues for more information:
     * - https://github.com/oracle/node-oracledb/issues/1688
     * - https://github.com/oracle/node-oracledb/issues/1691
     **/
    config.externals.push(
      ...[
        "@azure/app-configuration",
        "@azure/identity",
        "@azure/keyvault-secrets",
        "oci-common",
        "oci-objectstorage",
        "oci-secrets",
      ],
    );
	config.resolve.preferRelative = true;

    return config;
  },};

export default nextConfig;

```

The `config.resolve.preferRelative = true;` is used to tell Next.js to search relative paths as well.

**Comment by saphitv** [comment]
perfefct, thanks

**Comment by acid24** [comment]
Thanks @cjbj I'll give that a try

**Comment by sharadraju** [comment]
This is now fixed in the 6.7.1 release
